lib/services/alexa/request/event/runtime.ts

```diff
@@ -38,26 +37,28 @@ export const getVariable = (path: string, data: any) => {
   return !isNaN(curData) && curData.length < 16 ? +curData : curData;
 };
 
-export const _getEvent = (runtime: AlexaRuntime, extractFrame: typeof extractFrameCommand) => {
+export const getEvent = (runtime: AlexaRuntime) => {
   const request = runtime.getRequest();
 
   if (request?.type !== RequestType.EVENT) return null;
 
   const { event } = request.payload;
 
-  const res = extractFrame<EventCommand>(runtime.stack, (command: EventCommand | null) => command?.event === event);
+  const frames = runtime.stack.getFrames();
+  for (let index = frames.length - 1; index >= 0; index--) {
+    const commands = frames[index]?.getCommands<EventCommand>();
 
-  if (!res) return null;
+    // eslint-disable-next-line no-restricted-syntax
+    for (const command of commands) {
+      if (command?.event === event) {
+        return { index, command, event };
+      }
+    }
```

Comment:
since we are using optional chaining in the line above, to avoid crashes, it's better to check if we have commands before iteration. Looks like we should update general-service as well

```suggestion
  if (!commands) {
    continue;
  }

    // eslint-disable-next-line no-restricted-syntax
    for (const command of commands) {
      if (command?.event === event) {
        return { index, command, event };
      }
    }
```